### PR TITLE
[12.0][MIG] stock_analytic

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
supersedes #222

### CODE REVIEW

File: stock_analytic/models/stock.py [hunk 1]
```diff
@@ -33,3 +33,10 @@ def _prepare_account_move_line(self, qty, cost,
                     'analytic_account_id': self.analytic_account_id.id,
                 })
         return res
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    analytic_account_id = fields.Many2one(
+        related='move_id.analytic_account_id')
```
Comment (rousseldenis) [inline stock_analytic/models/stock.py]: @aheficent readonly=True or this is wanted ?

**Comment by sergio-teruel** [comment]
Continue in progress?

**Comment by AaronHForgeFlow** [comment]
@sergio-teruel I think this can be reviewed. The thing is, this module passes the analytic account from the stock move to the account move line, but the module that passes the analytic account to the stock move from other models is not implemented here.

File: stock_analytic/tests/test_stock_picking.py [hunk 2]
```diff
@@ -85,12 +106,12 @@ def _create_picking(
         return picking
 
     def _confirm_picking_no_error(self, picking):
-        picking.action_confirm()
+        picking.action_assign()
```
Comment (rousseldenis) [inline stock_analytic/tests/test_stock_picking.py]: @aheficent Why this change does not affect state ?

File: stock_analytic/tests/test_stock_picking.py [hunk 3]
```diff
@@ -85,12 +106,12 @@ def _create_picking(
...
         self.assertEqual(picking.state, 'confirmed')
 
     def _force_assign_no_error(self, picking):
-        picking.force_assign()
```
Comment (rousseldenis) [inline stock_analytic/tests/test_stock_picking.py]: @aheficent As reading Odoo message : https://github.com/odoo/odoo/blob/12.0/addons/stock/models/stock_picking.py#L705

IMHO You need to adapt quantities to simulate the former 'force'. You change the test behaviour here.

File: stock_analytic/tests/test_stock_picking.py (hunk 2 above)
Comment (AaronHForgeFlow) [reply stock_analytic/tests/test_stock_picking.py]: I need to recheck because I don't remember well; but I think action_confirm does not exist in v12 and assigning will auto confirm the picking if stock available

Comment (rousseldenis) [reply stock_analytic/tests/test_stock_picking.py]: https://github.com/odoo/odoo/blob/12.0/addons/stock/models/stock_picking.py#L547

Comment (AaronHForgeFlow) [reply stock_analytic/tests/test_stock_picking.py]: I was wrong. Thanks!

**Comment by pedrobaeza** [comment]
Check Travis status and avoid merge commits.

**Comment by AaronHForgeFlow** [comment]
@BT-cserra if you agree you can propose a PR with your new tests to my branch. Regards.

**Comment by BT-cserra** [comment]
> @BT-cserra if you agree you can propose a PR with your new tests to my branch. Regards.

Hello @aheficent I don't have access to your branch, only to 12.0.

**Comment by pedrobaeza** [comment]
@BT-cserra you can put the pull request against @aheficent's branch instead of OCA's one, but if both agree, I can close this PR and reopen the one done by @BT-cserra

**Comment by AaronHForgeFlow** [comment]
@pedrobaeza  I prefer to keep this one. 
@BT-cserra  I will include you commit with the changes in the tests here if you don't mind.

**Comment by BT-cserra** [comment]
> @BT-cserra I will include you commit with the changes in the tests here if you don't mind.
@aheficent OK.

**Comment by AaronHForgeFlow** [comment]
done in https://github.com/OCA/account-analytic/pull/223/commits/d2475c90c0515481ba9142818896c3a7c5f685d7

**Review by rousseldenis — APPROVED** [review]
Code review

**Review by MiquelRForgeFlow — APPROVED** [review]
